### ISSUE TITLE
gwd 7 crashes or can't find bases behind CGI under Windows

### ISSUE THREAD

**Issue by a2line** [geneweb/geneweb#12]
Under Apache 2.4 (Windows 7 x64), I have a dir /cgi-bin/ like this:
gwd6.exe > 6.07
gwd6.arg
gwd6b.exe > 6.08 (“new” one, with Ocaml 4.02.3)
gwd6b.arg
gwd7a.exe > 7.00 alpha
gwd7a.arg
gwd7.exe > 7.00 last build from master
gwd7.arg

To simplify the test, I made all .arg strictly identical containing only “`-cgi`”. If it works, we see the page requesting for base name (http://localhost/cgi-bin/gwd.exe). And with this little .arg every versions works, but we need more in .arg to reach our databases. And there it is if I goes back to a .arg with definitions of paths for bases/hd like this one:
```
-hd
F:\GW7\gw
-bd
F:\GW7\bases
-cgi
```

6.07 and 7.00a work (Ocaml 3.12) but 6.08 and 7.00 crash (Ocaml 4.02) : when doing a request from the browser, gwd take more and more memory (growing very quickly, about 200 Mo/s, and it nearly freezes the computer if the task is not killed quickly).

If I try with quotes for directories for eg. `"F:\GW7\gw"` it doesn't crash anymore, but it still can't find the bases. On the contrary, previous versions can't find bases if we had those quotes…

**Comment by GuillaumeBrochu** [comment]
Can we expect to have the same bug on Linux and MacOSX?

**Comment by a2line** [comment]
No, it's working on Linux > http://demo.geneweb.tuxfamily.org/gw7/gwd?b=grimaldi

**Comment by a2line** [comment]
I have renamed the issue because it's not linked to CGI at all. The behavior is the same without `-cgi` launching the program from console in daemon mode. I add that it works with network paths. So there's something with Windows drives or with the colon (:) itself.

Works:
```
gw6 -bd "F:\gw6\bases" -hd "F:\gw6\gw"
gw6 -bd F:\gw6\bases -hd F:\gw6\gw
gw7 -bd \\d\gw7\bases -hd \\d\gw7\gw
```

Don't work:
```
gw7 -bd "F:\gw7\bases" -hd "F:\gw7\gw"
gw7 -bd F:\gw7\bases -hd F:\gw7\gw
```

**Comment by GuillaumeBrochu** [comment]
It's either a bug related to the ocaml version used for compilation (4.02), or a bug related to the fixes for ocaml 4.x.

For the latter case, presuming that the bug would be in src/gwd.ml (this is not necessarily the case...), we should look at:
https://github.com/geneweb/geneweb/commits/master/src/gwd.ml
https://github.com/geneweb/geneweb/commits/distrib-6-08-ocaml-4-xx/src/gwd.ml
and the only common and relevant commit is:
https://github.com/geneweb/geneweb/commit/723fb3f9e685646f326de4b41d4017a9624af7c8
